ccvm_simulators/problem_classes/boxqp/problem_instance.py

```diff
@@ -143,10 +157,18 @@ def load_instance(
                 for idx in range(0, problem_size):
                     rval_v[idx] = -torch.Tensor([float(line_data_v[idx])])
                 # Read in the q_matrix matrix
-                for idx, line in enumerate(lines[2:]):
+                for idx, line in enumerate(lines[2 : problem_size + 2]):
                     line_data = line.split("\n")[0].split(file_delimiter)
                     for j, value in enumerate(line_data[:problem_size]):
                         rval_q[idx, j] = -torch.Tensor([float(value)])
+
+                # Read the last line as an additional information
+                last_raw_data_line = lines[-1].split("\n")[0].split(file_delimiter)
```

Comment:
We'll need to be more explicit, since this line is optional. Otherwise, if the user leaves out the final line then optional_data will become populated with the last line of q_matrix.
```suggestion
                last_raw_data_line = lines[problem_size + 3].split("\n")[0].split(file_delimiter)
```

Reply:
Any suggestion for this?

Reply:
I think you may have forgotten to push this change?